dev/ci/ci-common.sh

```diff
@@ -105,10 +105,25 @@ git_download()
   local giturl="${!giturl_var}"
   local ref_var="${project}_CI_REF"
   local ref="${!ref_var}"
+  local ref_for_rebase="$ref" # might be distinct from $ref if we're in a submoudle
+  local submodule_folder_var="${project}_CI_SUBMODULE_FOLDER"
+  local submodule_folder="${!submodule_folder_var}"
 
   local ov_url=${overlays[${project}_URL]}
   local ov_ref=${overlays[${project}_REF]}
 
+  local in_subfolder=""
+
+  # if the submodule_folder is non-empty, we clone to project-PARENT and
+  # then symlink the submodule_folder to dest this allows project CI scripts
+  # to be transparent w.r.t. whether or not the project is cloned from
+  # a submodule / submodule_folder.
+  # we can't symlink until the folder exists though
+  if [[ -n "$submodule_folder" ]]; then
+    local project_dest="$dest"
+    local dest="${project_dest}-PARENT"
```

Comment:
I'm not sure about this -PARENT stuff
I tried `make ci-fiat_crypto DOWNLOAD_ONLY=1` locally and it cloned fiat crypto and its submodules a bunch of times.

Why aren't we cloning fiat_crypto once and symlinking the submodules to it? ie something that would work out to
~~~
cd _build_ci
git clone $fiat_crypto_url fiat_crypto
(cd fiat_crypto && git submodule update --init --recursive)
ln -s  "$(pwd)fiat_crypto/rupicola" rupicola
(etc)
~~~

Reply:
Sure, I can add another argument to `project` identifying the parent project so we know what folder to clone to

Reply:
Hm, this is actually pretty annoying.  We can't update the submodule if the `.git` directory doesn't exist in the parent repo.  The fundamental problem here is that the download logic is inherently local but here you're asking for the logic to be non-local across projects.  Let me see if I can work something out though.  (It's going to make the logic somewhat more complicated though.)

Reply:
Argh, even https://github.com/coq/coq/pull/18824/files is not sufficient, because the leaf dependencies are going to download the parent project, and the .git will be removed, but then it becomes impossible to reuse the artifact later to adjust the submodules.  I guess I can make it work locally though.

Reply:
I've pushed an update that should avoid duplicate downloads when possible, but may rework and simplify the logic here later

Reply:
@SkySkimmer I've reorganized the code a bit.  There's now a separate `subproject` function in `ci-basic-overlay.sh` which takes `<name> <parent project> <submodulefolder> <submodule giturl> <submodule branch>` as in `subproject kami fiat_crypto "rupicola/bedrock2/deps/kami" "https://github.com/mit-plv/kami" "rv32i"`.  There is no difference in declaring or submitting overlays, the parent repository should be downloaded only once (unless you have already deleted it's `.git` folder).

I've removed the documentation on submitting overlays to submodules, because I don't think it adds anything anymore, the process is not really different except that you read from the arguments to `subproject` instead of `project`.